src/Schema/AbstractSchema.php

```diff
@@ -394,24 +402,53 @@ protected function findTableNames(string $schema): array
     }
 
     /**
-     * Extracts the PHP type from an abstract DB type.
+     * Creates a column schema for the database.
      *
-     * @param ColumnSchemaInterface $column The column schema information.
+     * This method may be overridden by child classes to create a DBMS-specific column schema.
+     *
+     * @param string $type The abstract data type.
+     * @param string $name The column name.
+     * @param mixed ...$info The column information.
+     * @psalm-param array{unsigned?: bool} $info The set of parameters may be different for a specific DBMS.
+     *
+     * @return ColumnSchemaInterface
+     */
+    protected function createColumnSchema(string $type, string $name, mixed ...$info): ColumnSchemaInterface
+    {
+        $phpType = $this->getColumnPhpType($type);
+        $isUnsigned = !empty($info['unsigned']);
+
+        if (
+            $isUnsigned && PHP_INT_SIZE === 4 && $type === SchemaInterface::TYPE_INTEGER
+            || ($isUnsigned || PHP_INT_SIZE !== 8) && $type === SchemaInterface::TYPE_BIGINT
+        ) {
+            $column = new BigIntColumnSchema($name);
+        } else {
+            $column = $this->createPhpTypeColumnSchema($phpType, $name);
```

Comment:
Very strange behavior. Why BigIntColumn not in this function ?

Reply:
Creating `BigIntColumnSchema` depends on `$isUnsigned`, `PHP_INT_SIZE`, `$type`
`createPhpTypeColumnSchema()` depends only on `$phpType`

Reply:
Done